### ISSUE TITLE
Installation Guide

### ISSUE THREAD

**Issue by deadmantfa** [hiqdev/yii2-mfa#1]
Hi HiqDev Team,

This seems like a great project and I would like to use the same in mine.

I cant seem to wrap my head around how to set this up in my project, can you guys/gals update the readme with more information

Regards,
Wenceslaus Dsilva

**Comment by hiqsol** [comment]
Hi!

This plugin is installed the usual way: by requiring it with composer.
Then you need to configure this plugin. There are 2 options:

- use [composer-config-plugin] to load configuration automatically (this is simpler
for us, cause we are used to it)
- use old good normal way: copy-paste provided configuration from [src/config/web.php] into configuration
of your application

[composer-config-plugin]:   https://github.com/hiqdev/composer-config-plugin

After adding configuration you are basically done, cause plugin provides **behavior** 
and configuration attaches it to **user** component on beforeLogin event.
And then the behavior validates ips and TOTP on every login.

You need to have `allowed_ips` readable and writable property in your **user** component for allowed IPs functionality.
And `totp_secret` property for TOTP functionality.

So, to use this plugin:

- install it with composer
- add configuration to your application
- provide `allowed_ips` and totp_secret` properties in **user** component

Basically that's everything to describe about the plugin in general.
Maybe we could help you more if you have more specific questions.

**Comment by norfarahim** [comment]
Hi @hiqsol ! I'm really new to Yii2. So maybe I'm not quite understand with the guide but I still try tho. So here I want to ask you. For the `allowed_ips` & `totp_secret`, is it need to put in common/model/user? If so is it like this?
![image](https://user-images.githubusercontent.com/34227840/37575543-b14eccd4-2b62-11e8-9f57-41365c593b00.png)
But I got an error for that. Hope you can help 😅. Thank you !

**Comment by hiqsol** [comment]
Yes, `User` model is the right place.
What kind of error do you receive? Copy your error message for I could have an idea.

**Comment by norfarahim** [comment]
Thank you for the reply ! Here is the error 
![image](https://user-images.githubusercontent.com/34227840/37589299-c394d752-2b9f-11e8-9c55-0734cde66045.png)
Do you have an idea what's wrong with it?

**Comment by hiqsol** [comment]
Thank you for the report. That was PHP 7.2 incompatibility.
Fixed with https://github.com/hiqdev/yii2-mfa/commit/a2eb7a95f14ed583855d26eac6563675391fdea6
Update and try again.

**Comment by norfarahim** [comment]
Thanks! Don't have the error now. Btw, is the theme-manager need to use or optional?

**Comment by hiqsol** [comment]
Yii2-theme-manager is required now.
